modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/display/context/JournalEditDDMStructuresDisplayContext.java

```diff
@@ -46,14 +53,36 @@
 public class JournalEditDDMStructuresDisplayContext {
 
 	public JournalEditDDMStructuresDisplayContext(
-		HttpServletRequest httpServletRequest,
-		LiferayPortletResponse liferayPortletResponse) {
+		Portal portal, RenderRequest renderRequest,
+		RenderResponse renderResponse) {
+
+		_httpServletRequest = portal.getHttpServletRequest(renderRequest);
+		_liferayPortletResponse = portal.getLiferayPortletResponse(
+			renderResponse);
+
+		_themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		JournalServiceConfiguration journalServiceConfiguration;
+
+		try {
+			journalServiceConfiguration =
+				ConfigurationProviderUtil.getCompanyConfiguration(
+					JournalServiceConfiguration.class,
+					_themeDisplay.getCompanyId());
+		}
+		catch (ConfigurationException configurationException) {
+			journalServiceConfiguration = null;
+
+			if (_log.isDebugEnabled()) {
+				_log.debug(configurationException);
+			}
+		}
```

Comment:
Why are you ignoring the error? Unless there's a good reason, it is better to let if fail.

Reply:
Also, the variable is not necessary; you can assign the configuration directly to the field.

Reply:
@adolfopa why do you prefer throw the exception and broke the application instead of let them work without configuration?

Reply:
In this case the use of the configuration is to enable user to modify DDM structure key and for me it is better to let the user work to create DDM instead of broke the ddm form.

Reply:
Have you seen it fail? The configuration has default values for all fields, so ConfigurationProviderUtil should not return null and at least the default values should be available.

Reply:
No. I have not seen it fail but just in case only but, ok, no problem, I will change it. I like and for me it was safer to have it like this but I do not have any problem to change it. So I will change it.